### DISCUSSION TITLE
AOSP 12.1 v410

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2219] [General]
- Android 12.1. This is considered a pre-release because it's the first Android 12.1 build. As such, it won't be available as OTA.
**- Warning: gapps variant has broken Launcher**
- March security patch
- Fix touchscreen on some Samsung devices
- Fix audio on Moto One Fusion+ (Thanks @ksdrar)
- Disable fingerprint gestures on LG Stylo 7 (Thanks @CHIDILights)
- Disable fingerprint gestures on Doogee S88 Pro
- Map camera/focus buttons on all Sony devices
- Fix setup wizard in arm32_binder64 gapps
- Enable Widevine L1 on Xiaomi Pad 5
- Fix modem on Redmi 9A
- Fix WPA3 by disabling it
- Remove christmas boot animation (I'm open to new boot animations!)
- Fix 120hz/90hz on Xiaomi Civi (thanks @AndyCGYan)
- Fix under-display fingerprint on Asus Zenfone8
- Add an option in updater settings to uninstall OTA

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v410'>AOSP 12.1 v410</a>.</em>

**Comment by robato22** [comment]
Can I install this in my Blu Vivo XL4?


OS | Android 8.1 (Oreo)
-- | --
Chipset | Mediatek MT6762 Helio P22 (12 nm)
CPU | Octa-core 2.0 GHz Cortex-A53
GPU | PowerVR GE8320

**Reply by tnolte1** [reply]
I have a Blackview A80Plus with the same SOC (MT Helio P22) and it works just fine.

**Comment by tnolte1** [comment]
Thumbs up for OTA uninstall option 👍

**Reply by steupz** [reply]
>  👍

Any issues with bluetooth audio

**Comment by iviliprins** [comment]
There is no animation in "lift to check phone animation"
Can fix?

**Comment by der-lange** [comment]
My device:  Sony Xperia XZ2P (H8166_Customized DACHBLX_52.1.A.3.49-R6C)
system-image: [system-squeak-arm64-ab-vndklite-gogapps.img
Installation: adb reboot bootloader;  fastboot erase system_a; 
                       fastboot --disable-verity --disable-verification flash vbmeta vbmeta.img;
                       fastboot flash system_a system-xxxx.img
 Results:         version start correctly; but  while restoring my apps from google-drive and reentering the PIN for safe the app stops
                         "Camera error, Cant connect the camera"

**Reply by der-lange** [reply]
Cameraproblem
I found this:
"there won't be an official support, because the vendor of the bionz camera chip didn't allow the usage"
What does it mean for the new AOSP?
Is it possible to find another solution (GCAM)?

**Reply by devrieshh** [reply]
GCam_Go.GreatnessTest2.apk  normaly this one works , or just    GCam_Go.apk

**Reply by der-lange** [reply]
thanks for your advice! But: no chance.
Camera keeps stopping; close the device
or
something went wrong, try again

**Comment by devrieshh** [comment]
You used Google camera go?

**Reply by der-lange** [reply]
Thank you for your advices. For me I found a "perfect" solution:
https://forum.xda-developers.com/t/xz2p-on-gsi-11-12-android.4421169/#post-86649107
Even the original SONY Xperia camera is working fine under android 12. Have a look at this link. It´s worth while!

**Reply by phhusson** [reply]
@der-lange Camera is fixed on Xperia XZ2 Premium, simply by changing ro.build.fingerprint to stock fingerprint?

**Reply by der-lange** [reply]
@phhusson No, completely installed CorvusOS + magisk + permissiver + camera-0.3.0-tama.zip.
My knowledge about these things is poor. For more details, I think it´s better to contact the developper @Kasper666666

**Comment by Cerebral20** [comment]
To fix recovery stuck, in Exynos 7870 Devices-

can you please add this to build prop of system
persist.sys.binary_xml=false

**Comment by nation888** [comment]
Oukitel K9 MT6765 .. 
Previous versions worked no problem. This version; bootloop. TWRP 3.3.1 by mechanik6. I'm not sure what I'm missing. Anyone else?

**Comment by Oldbrahman** [comment]
屏下指纹能用吗？

**Comment by fatmalama** [comment]
compatible launchers for the gapps version? I want to try the new tablet nav
(anyone reading this now: try the newer releases)

**Reply by Oldbrahman** [reply]
Screen fingerprint unlock

**Reply by fatmalama** [reply]
what??? I have screen lock disabled because I thought maybe that was bugging the nav but no

**Comment by IverCoder** [comment]
Isn't it 12L, not 12.1... but anyway great work, helpful in giving a new life to rare devices 🥳

**Reply by phhusson** [reply]
12L is Google marketing name for 12.1, see https://android.googlesource.com/platform/manifest/+/refs/tags/android-12.1.0_r1 it does say 12.1

**Comment by Aj-style** [comment]
is camera HAL v1 back ? cause my camera is broken on android 12 aosp gsi v402

**Comment by retardedloozer** [comment]
My phone is a Moto E6. I installed both the arm32_binder64_ab_gapps and a64_ab_gapps_vndklite roms and while they certainly boot, the recents button does not work. The bluetooth audio doesn't work either but I need the recent button more than I need bluetooth audio. I searched the issues section and I have seen that I'm not the only one with the issue. I saw the adb shell commands in one of the issue threads and did issue them but it didn't work. Recents button works on the vanilla version, but I also can't flash gapps. This makes for an issue. But I understand that this is a pre-release so I'm just leaving this comment for bug reporting purposes. Thank you for your service to the modding community

**Reply by mariasanchez84** [reply]
use the launcher from vanilla version

**Comment by mariasanchez84** [comment]
Awesome, works great thanks a lot !

**Comment by farmass** [comment]
my device is a redmi 9a, android 12.1 has lag in the animations, it's very annoying... I see that the compilation is new, thanks 😁
